File: eCinema.Services/Database/Kupovina.cs
using System;
using System.Collections.Generic;

namespace eCinema.Services.Database
{
    public partial class Kupovina
    {
        public int KupovinaId { get; set; }
        public int Kolicina { get; set; }
        public int Cijena { get; set; }
        public DateTime? DatumKupovine { get; set; }
        public string? PaymentIntentId { get; set; }
        public bool? Placena { get; set; }
        public int? KorisnikId { get; set; }
        public virtual Korisnik? Korisnik { get; set; }
        public int? TerminId { get; set; }
        public virtual Termin? Termin { get; set; }
    }
}

File: ecinema_mobile/lib/models/kupovina.dart
import 'package:ecinema_mobile/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'kupovina.g.dart';

@JsonSerializable()
class Kupovina {
  int kupovinaId;
  int? kolicina;
  int? cijena;
  DateTime datumKupovine;
  String? paymentIntentId;
  int? korisnikId;
  Korisnik? korisnik;
  int? terminId;
  Termin? termin;
  bool? placena;

  Kupovina(
      {required this.kupovinaId,
      this.kolicina,
      this.cijena,
      required this.datumKupovine,
      this.paymentIntentId,
      this.korisnikId,
      this.korisnik,
      this.terminId,
      this.termin,
      this.placena});

  factory Kupovina.fromJson(Map<String, dynamic> json) =>
      _$KupovinaFromJson(json);
  Map<String, dynamic> toJson() => _$KupovinaToJson(this);
}
